test/registration/ui/addon_eula_dialog_test.rb

```diff
@@ -15,13 +15,129 @@
 require "registration/ui/addon_eula_dialog"
 
 describe Registration::UI::AddonEulaDialog do
-  subject(:dialog) { described_class.new([addon]) }
+  subject(:dialog) { described_class.new(addons) }
 
-  let(:addon) do
-    addon_generator("name" => "SLES", "eula_url" => "https://suse.com/download/SLES/eula")
+  let(:params) { { "eula_url" => "http://example.addon-eula.url" } }
+  let(:addons) { [registered_addon, addon_wo_eula, addon_with_eula] }
+
+  let(:addon_wo_eula) { Registration::Addon.new(addon_generator) }
+  let(:addon_with_eula) { Registration::Addon.new(addon_generator(params)) }
+  let(:second_addon_with_eula) { Registration::Addon.new(addon_generator(params)) }
+  let(:registered_addon) { Registration::Addon.new(addon_generator(params)) }
+
+  let(:product_license) do
+    instance_double(Y2Packager::ProductLicense, accepted?: false, accept!: true)
+  end
+
+  describe "#run" do
+    before do
+      allow(Yast::Wizard).to receive(:SetContents)
+      allow(dialog).to receive(:find_license).and_return(product_license)
+      allow(dialog).to receive(:download_eula).and_return(true)
+      registered_addon.registered
+    end
+
+    context "when there are no EULA acceptances to show" do
+      let(:addons) { [registered_addon, addon_wo_eula] }
+
+      it "does not display the eula dialog" do
```

Comment:
NP: the casing should be unified
```suggestion
      it "does not display the EULA dialog" do
```